### PR TITLE
review trick

### PR DESCRIPTION
**Pull request by calvin-dani**

### CODE REVIEW

File: blockonomics-billboard/src/setupTests.js [hunk 1]
```diff
@@ -1,5 +0,0 @@
-// jest-dom adds custom jest matchers for asserting on DOM nodes.
```
Comment (chrisdani-dev) [inline blockonomics-billboard/src/setupTests.js:1]: why are these comments here?

File: blockonomics-analytics-server/.ini [hunk 2]
```diff
@@ -1,5 +0,0 @@
-[PROD]
-DB_URI = TO_BE_ADDED_BY_USERS_MONGODB_CLUSTER_PASSWORD
```
Comment (chrisdani-dev) [inline blockonomics-analytics-server/.ini:2]: why is this to be added here?

File: blockonomics-analytics-server/.ini [hunk 3]
```diff
@@ -1,5 +0,0 @@
...
-PORT = 5005
-[TEST]
-DB_URI = your_testing_db_uri (can be the same as Atlas, or a local MongoDB database)
```
Comment (chrisdani-dev) [inline blockonomics-analytics-server/.ini:5]: why is this here?

File: blockonomics-analytics-server/analytical_server.py [hunk 4]
```diff
@@ -1,38 +0,0 @@
...
-from db_analytics import get_comment_largest_value
-from pymongo import MongoClient
-from flask_apscheduler import APScheduler
-from flask_cors import CORS, cross_origin
-from flask_caching import Cache
-
-app2 = Flask(__name__)
-cache = Cache(app2)
-app2.config['CORS_HEADERS'] = "Content-Type"
-ALLOWED_ORIGINS = ['localhost', '127.0.0.1']
```
Comment (chrisdani-dev) [inline blockonomics-analytics-server/analytical_server.py:14]: constants should be higher

File: blockonomics-analytics-server/analytical_server.py [hunk 5]
```diff
@@ -1,38 +0,0 @@
...
-
-cors = CORS(app2, resources={"/*": {"origins": ALLOWED_ORIGINS}})
-
-config = configparser.ConfigParser()
-config.read(os.path.abspath(os.path.join(".ini")))
-result = None
-
-#Get max valued message
-@app2.route("/get_message",methods=["GET"])
-@cross_origin()
```
Comment (chrisdani-dev) [inline blockonomics-analytics-server/analytical_server.py:24]: why you need cross origin?

File: blockonomics-analytics-server/analytical_server.py [hunk 6]
```diff
@@ -1,38 +0,0 @@
...
-app2.config['CORS_HEADERS'] = "Content-Type"
-ALLOWED_ORIGINS = ['localhost', '127.0.0.1']
-
-cors = CORS(app2, resources={"/*": {"origins": ALLOWED_ORIGINS}})
-
-config = configparser.ConfigParser()
-config.read(os.path.abspath(os.path.join(".ini")))
-result = None
-
-#Get max valued message
```
Comment (chrisdani-dev) [inline blockonomics-analytics-server/analytical_server.py:22]: why there is no space after # ? and the description should be more clear. as in what is message , what is compared to get max etc?

File: blockonomics-analytics-server/analytical_server.py [hunk 7]
```diff
@@ -1,38 +0,0 @@
...
-from flask_cors import CORS, cross_origin
-from flask_caching import Cache
-
-app2 = Flask(__name__)
-cache = Cache(app2)
-app2.config['CORS_HEADERS'] = "Content-Type"
-ALLOWED_ORIGINS = ['localhost', '127.0.0.1']
-
-cors = CORS(app2, resources={"/*": {"origins": ALLOWED_ORIGINS}})
-
```
Comment (chrisdani-dev) [inline blockonomics-analytics-server/analytical_server.py:17]: comment on top saying what purpose this file serves? Ex. Here we have all the functions to serve the api.

File: blockonomics-analytics-server/analytical_server.py [hunk 8]
```diff
@@ -1,38 +0,0 @@
...
-import requests
-import os
-import configparser
-from db_analytics import get_comment_largest_value
-from pymongo import MongoClient
-from flask_apscheduler import APScheduler
-from flask_cors import CORS, cross_origin
-from flask_caching import Cache
-
-app2 = Flask(__name__)
```
Comment (chrisdani-dev) [inline blockonomics-analytics-server/analytical_server.py:11]: i feel all of this should be part of some init function for this file

File: blockonomics-analytics-server/analytical_server.py [hunk 9]
```diff
@@ -1,38 +0,0 @@
...
-#Get max valued message
-@app2.route("/get_message",methods=["GET"])
-@cross_origin()
-def get_message_max_value():
-    with app2.app_context():
-        result = get_comment_largest_value()
-        if result:
-            return result,200
-        else:
-            return result,500
```
Comment (chrisdani-dev) [inline blockonomics-analytics-server/analytical_server.py:31]: why we return result if no result? and what is 500 ,  is it http error code?

File: blockonomics-analytics-server/analytical_server.py [hunk 10]
```diff
@@ -1,38 +0,0 @@
...
-
-if __name__ == '__main__':
-    # run app in debug mode on port 5005
```
Comment (chrisdani-dev) [inline blockonomics-analytics-server/analytical_server.py:34]: write an example on how to run this file. like. run analytical_server.py --debug or whatever is relevant

File: blockonomics-analytics-server/db_analytics.py [hunk 11]
```diff
@@ -1,35 +0,0 @@
-from flask import current_app, g
-from werkzeug.local import LocalProxy
-from flask_pymongo import PyMongo
-from bson.objectid import ObjectId
-from flask.json import JSONEncoder
-from bson import json_util, ObjectId
-from datetime import datetime
-
-
-def get_db():
```
Comment (chrisdani-dev) [inline blockonomics-analytics-server/db_analytics.py:10]: where is comment?

File: blockonomics-analytics-server/db_analytics.py [hunk 12]
```diff
@@ -1,35 +0,0 @@
...
-    Configuration method to return db instance
-    """
-    db = getattr(g, "_database", None)
-
-    if db is None:
-        db = g._database = PyMongo(current_app).db
-    return db
-
-# Use LocalProxy to read the global db instance with just `db`
-db = LocalProxy(get_db)
```
Comment (chrisdani-dev) [inline blockonomics-analytics-server/db_analytics.py:21]: this should be inside a funcion

File: blockonomics-analytics-server/db_analytics.py [hunk 13]
```diff
@@ -1,35 +0,0 @@
...
-
-#Querying the max valued message
```
Comment (chrisdani-dev) [inline blockonomics-analytics-server/db_analytics.py:23]: space

File: blockonomics-analytics-server/db_analytics.py [hunk 14]
```diff
@@ -1,35 +0,0 @@
...
-def get_comment_largest_value():
-    result = db.Donation_msg.find().sort("value",-1).limit(1)[0]
```
Comment (chrisdani-dev) [inline blockonomics-analytics-server/db_analytics.py:25]: what if nothign is returned, is it always guaranteed to have atleast one value in return result set?

File: blockonomics-analytics-server/db_analytics.py [hunk 15]
```diff
@@ -1,35 +0,0 @@
...
-    return json_util.dumps(result) if result else {"message":"","name":""}
```
Comment (chrisdani-dev) [inline blockonomics-analytics-server/db_analytics.py:26]: make it like a struct instead of writing here "message" and "name"

File: blockonomics-analytics-server/db_analytics.py [hunk 16]
```diff
@@ -1,35 +0,0 @@
...
-
-class MongoJsonEncoder(JSONEncoder):
```
Comment (chrisdani-dev) [inline blockonomics-analytics-server/db_analytics.py:28]: i think this should be in separate file.

File: blockonomics-analytics-server/db_analytics.py [hunk 17]
```diff
@@ -1,35 +0,0 @@
...
-
-    def default(self, obj):
```
Comment (chrisdani-dev) [inline blockonomics-analytics-server/db_analytics.py:30]: where comment

File: blockonomics-server/crud_server.py [hunk 18]
```diff
@@ -1,47 +0,0 @@
...
-
-#binding Flask application  
-app1 = Flask(__name__)
-
-#Fetching values from .ini files
-config = configparser.ConfigParser()
-config.read(os.path.abspath(os.path.join(".ini")))
-
-
-#Open to blockonomics as a webhook endpoint
```
Comment (chrisdani-dev) [inline blockonomics-server/crud_server.py:17]: not clear

File: blockonomics-server/crud_server.py [hunk 19]
```diff
@@ -1,47 +0,0 @@
...
-    #Check if payment is successfull
-    if return_status(status):
-        #Fetch the details of the transaction
-        endpoint, headers = set_endpoint_metadata(uuid)
-        payload = requests.get(endpoint, headers=headers).json()
-        #Parsing response
-        data = parse_transaction_payload(payload)
-        with app1.app_context():
-            #Add the transaction detalils to mongodb
-            add_comment(data)
```
Comment (chrisdani-dev) [inline blockonomics-server/crud_server.py:31]: why is it called add_comment?

File: blockonomics-server/db.py [hunk 20]
```diff
@@ -1,71 +0,0 @@
...
-from pymongo.errors import DuplicateKeyError, OperationFailure
-from bson.objectid import ObjectId
-from bson.errors import InvalidId
-
-import os
-
-from flask import Flask, render_template
-from flask.json import JSONEncoder
-from flask_cors import CORS
-##from flask_bcrypt import Bcrypt
```
Comment (chrisdani-dev) [inline blockonomics-server/db.py:16]: y

File: blockonomics-server/db.py [hunk 21]
```diff
@@ -1,71 +0,0 @@
...
-##from flask_jwt_extended import JWTManager
-
-from bson import json_util, ObjectId
-from datetime import datetime, timedelta
-
-
-def get_db():
```
Comment (chrisdani-dev) [inline blockonomics-server/db.py:23]: comment

File: blockonomics-server/db.py [hunk 22]
```diff
@@ -1,71 +0,0 @@
...
-        db = g._database = PyMongo(current_app).db
-
-    return db
-
-
-# Use LocalProxy to read the global db instance with just `db`
-db = LocalProxy(get_db)
-
-
-def add_comment(data):
```
Comment (chrisdani-dev) [inline blockonomics-server/db.py:40]: comment

File: blockonomics-server/db.py [hunk 23]
```diff
@@ -1,71 +0,0 @@
...
-        'sender_email': data["emailid"],
-        'message': data["Message"],
-        'value': data["paid_satoshi"],
-        'timestamp': data["timestamp"],
-    }
-    return db.Donation_msg.insert_one(comment_doc)
-
-
-
-class MongoJsonEncoder(JSONEncoder):
```
Comment (chrisdani-dev) [inline blockonomics-server/db.py:62]: diff file

File: blockonomics-server/payload_utils.py [hunk 24]
```diff
@@ -1,43 +0,0 @@
-
-
-#Status of transaction and proceed map
-status_order = {'2': True, '1': False, '0': False}
```
Comment (chrisdani-dev) [inline blockonomics-server/payload_utils.py:4]: should be in upper case

File: blockonomics-server/payload_utils.py [hunk 25]
```diff
@@ -1,43 +0,0 @@
...
-
-#Parses from blockonomics
-def parse_transaction_payload(payload):
-    data = {}
-    for attr in important_attr:
-        try:
-            data[attr] = get_payload_attr(payload, attr, attr in attr_data_arr)
-        except Exception as e:
-            data[attr] = ""
-            print(e)
```
Comment (chrisdani-dev) [inline blockonomics-server/payload_utils.py:27]: y print

File: blockonomics-server/payload_utils.py [hunk 26]
```diff
@@ -1,43 +0,0 @@
...
-        else:
-            continue
-    return data
-
-#Flattens the response
-def get_payload_attr(payload, attr, data_arr_req=False):
```
Comment (chrisdani-dev) [inline blockonomics-server/payload_utils.py:33]: what response

File: blockonomics-server/tests.py [hunk 27]
```diff
@@ -1,44 +0,0 @@
...
-    def test_endpoints(self):
-        url, obj = set_endpoint_metadata("1234")
-        self.assertEqual(
-            url, "https://www.blockonomics.co/api/merchant_order/1234")
-
-    #Test utility parser to retrieve transaction details 
-    def test_payload_creator(self):
-        
-        important_attr = {
-            "uuid", "address", "name", "emailid", "Message", "paid_satoshi",
```
Comment (chrisdani-dev) [inline blockonomics-server/tests.py:21]: why message is camel case?

File: blockonomics-server/tests.py [hunk 28]
```diff
@@ -1,44 +0,0 @@
...
-            "timestamp"
-        }
-
-        attr_data_arr = {"name", "emailid", "Message"}
-
-        data = {}
-        payload = {"data": {}}
-        for attr in important_attr:
-            if not (attr in attr_data_arr):
```
Comment (chrisdani-dev) [inline blockonomics-server/tests.py:30]: i think you should write, if attr not in attr_data_arr

Comment (chrisdani-dev) [inline blockonomics-server/tests.py:30]: also i think all this should be in another function called validate

File: blockonomics-analytics-server/analytical_server.py (hunk 5 above)
Comment (calvin-dani) [reply blockonomics-analytics-server/analytical_server.py:24]: currently nothing is deployed in cloud except for the CRUD server. Analytics server is in localhost and need cross origin

File: blockonomics-analytics-server/analytical_server.py (hunk 9 above)
Comment (calvin-dani) [reply blockonomics-analytics-server/analytical_server.py:31]: yes its error code, result is changed

File: blockonomics-billboard/public/index.html [hunk 29]
```diff
@@ -1,44 +0,0 @@
...
-      work correctly both with client-side routing and a non-root public URL.
-      Learn how to configure a non-root public URL by running `npm run build`.
-    -->
-    <title>React App</title>
-  </head>
-  <body>
-    <noscript>You need to enable JavaScript to run this app.</noscript>
-    <div id="root"></div>
-    <!--
-      This HTML file is a template.
```
[the remark this answers is not in the document]
Comment (calvin-dani) [reply blockonomics-billboard/public/index.html:33]: Yes this file is needed

File: blockonomics-billboard/src/App.js [hunk 30]
```diff
@@ -1,64 +0,0 @@
...
-   
-    //Ping the analytic server for the latest message of the highest donor
-    let interval = setInterval(get_message, 30000); // set the interval time in milliseconds
-
-    return () => clearInterval(interval); // clean up the interval when unmounting
-  }, []);
-
- 
-  return (
-    <div className="App">
```
[the remark this answers is not in the document]
Comment (calvin-dani) [reply blockonomics-billboard/src/App.js:41]: main root component

File: blockonomics-server/.ini [hunk 31]
```diff
@@ -1,8 +0,0 @@
-[PROTO]
-GET_TRANSACTION_URL = https://www.blockonomics.co/api/merchant_order/
-DB_URI = TO_BE_ADDED_BY_USERS_MONGODB_CLUSTER
```
[the remark this answers is not in the document]
Comment (calvin-dani) [reply blockonomics-server/.ini:3]: Secret info to be added by users

File: blockonomics-server/db.py (hunk 20 above)
Comment (calvin-dani) [reply blockonomics-server/db.py:16]: ?

File: blockonomics-server/payload_utils.py (hunk 25 above)
Comment (calvin-dani) [reply blockonomics-server/payload_utils.py:27]: debug purposes, exception is handled in else statement

File: blockonomics-server/tests.py (hunk 27 above)
Comment (calvin-dani) [reply blockonomics-server/tests.py:21]: that how I get it from api

File: blockonomics-server/tests.py (hunk 28 above)
Comment (calvin-dani) [reply blockonomics-server/tests.py:30]: in test ?

File: blockonomics-analytics-server/analytical_server.py (hunk 8 above)
Comment (calvin-dani) [reply blockonomics-analytics-server/analytical_server.py:11]: some errors came up when I tried another way? this works fine and its from documentation. Ill check why later

File: blockonomics-analytics-server/db_analytics.py (hunk 12 above)
Comment (calvin-dani) [reply blockonomics-analytics-server/db_analytics.py:21]: its a weird concept in python leaving it untouched, not sure how it behaves differently

File: blockonomics-analytics-server/db_analytics.py (hunk 14 above)
Comment (calvin-dani) [reply blockonomics-analytics-server/db_analytics.py:25]: There's always one value in return set

File: blockonomics-analytics-server/.ini (hunk 2 above)
Comment (chrisdani-dev) [reply blockonomics-analytics-server/.ini:2]: yes, but your use case does not need it right?

File: blockonomics-analytics-server/analytical_server.py (hunk 8 above)
Comment (chrisdani-dev) [reply blockonomics-analytics-server/analytical_server.py:11]: ok, organize is so that constants are together, then white space then others

File: blockonomics-analytics-server/analytical_server.py (hunk 5 above)
Comment (chrisdani-dev) [reply blockonomics-analytics-server/analytical_server.py:24]: why you cannot use both in cloud? why one here one there ?

File: blockonomics-analytics-server/analytical_server.py (hunk 9 above)
Comment (chrisdani-dev) [reply blockonomics-analytics-server/analytical_server.py:31]: then you should use https package and refer it like https.internal_error or something like that, you should not hard code

File: blockonomics-analytics-server/db_analytics.py (hunk 12 above)
Comment (chrisdani-dev) [reply blockonomics-analytics-server/db_analytics.py:21]: no initialization should be open, everything should be in a function and it should be called correctly. like this can be in def initialize_db(); and then you call it somewhere first.

File: blockonomics-analytics-server/db_analytics.py (hunk 13 above)
Comment (chrisdani-dev) [reply blockonomics-analytics-server/db_analytics.py:23]: why closed without making space?

File: blockonomics-analytics-server/db_analytics.py (hunk 14 above)
Comment (chrisdani-dev) [reply blockonomics-analytics-server/db_analytics.py:25]: write a comment , saying this query does this

File: blockonomics-analytics-server/db_analytics.py (hunk 15 above)
Comment (chrisdani-dev) [reply blockonomics-analytics-server/db_analytics.py:26]: why you resolved without making the change?

File: blockonomics-analytics-server/db_analytics.py (hunk 16 above)
Comment (chrisdani-dev) [reply blockonomics-analytics-server/db_analytics.py:28]: why its closed without moving to another file

File: blockonomics-analytics-server/db_analytics.py (hunk 17 above)
Comment (chrisdani-dev) [reply blockonomics-analytics-server/db_analytics.py:30]: where is the comment? why did you close without having a comment?

File: blockonomics-billboard/.gitignore [hunk 32]
```diff
@@ -1,23 +0,0 @@
-# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
```
[the remark this answers is not in the document]
Comment (chrisdani-dev) [reply blockonomics-billboard/.gitignore:1]: why you resolved without addressing comment?

File: blockonomics-server/tests.py (hunk 28 above)
Comment (chrisdani-dev) [reply blockonomics-server/tests.py:30]: you tell

File: blockonomics-analytics-server/analytical_server.py (hunk 5 above)
Comment (calvin-dani) [reply blockonomics-analytics-server/analytical_server.py:24]: ngrok free version only allows 1 to be tunneled to cloud. Didn't want to use EC2 instance as my blockonomics account might get blocked

Comment (calvin-dani) [reply blockonomics-analytics-server/analytical_server.py:24]: So used ngrok to deploy the one server which is exposed for the web hook to connect
